homeassistant/components/ring/switch.py

```diff
@@ -0,0 +1,115 @@
+"""This component provides HA switch support for Ring Door Bell/Chimes."""
+from datetime import timedelta
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+
+from . import DATA_RING
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=10)
+
+SIREN_ICON = 'mdi:alarm-bell'
+LIGHT_ICON = 'mdi:track-light'
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a sensor for a Ring device."""
+    ring = hass.data[DATA_RING]
+    logging.warning('setting up')
+    switches = []
+
+    for device in ring.stickup_cams:  # ring.stickup_cams is doing I/O
+        if device.has_capability('siren'):
+            switches.append(SirenSwitch(hass, device))
+        if device.has_capability('light'):
+            switches.append(LightSwitch(hass, device))
+
+    add_entities(switches, True)
+
+
+class BaseRingSwitch(SwitchDevice):
+    """Represents a switch for controlling an aspect of a ring device"""
```

Comment:
Represent a switch...

Please end with period.

Reply:
I used flake8 - rest should be fine I hope!